src/MiniSequenceEndorser.sol

```diff
@@ -535,4 +560,230 @@ contract MiniSequenceEndorser is IEndorser, Owned {
 
   }
 
+  function operationFilter(
+    IEndorser.Operation calldata operation
+  ) external view returns (
+    address _address,
+    bytes32[][] memory topics
+  ) {
+    _address = operation.entrypoint;
+
+    ExecuteCall memory call = _decodeExecuteCall(operation.data);
+
+    if (_address == GUEST_MODULE_V1) {
+      // TODO: There's no possible filter that can detect v1 TxExecuted events.
+    } else if (_address == GUEST_MODULE_V2) {
+      bytes32 digest = keccak256(abi.encode('guest:', call.txs));
+      bytes32 subdigest = keccak256(abi.encodePacked("\x19\x01", block.chainid, _address, digest));
+
+      topics = new bytes32[][](2);
+      topics[0] = new bytes32[](2);
+      topics[0][0] = TX_EXECUTED_V2;
+      topics[0][1] = TX_FAILED_V2;
+      topics[1] = new bytes32[](1);
+      topics[1][0] = subdigest;
+
+      return (_address, topics);
+    } else {
+      address implementation = abi.decode(operation.endorserCallData, (address));
+
+      if (implementation == MAIN_MODULE_V1 || implementation == MAIN_MODULE_UPGRADABLE_V1) {
+        // TODO: There's no possible filter that can detect v1 TxExecuted events.
```

Comment:
Nonce change + space? it could lead to false positives if the transaction is replaced, but it is better than nothing, right? IDK if it is worth it